Mathlib/Algebra/Order/Archimedean/Basic.lean

```diff
@@ -523,3 +523,26 @@ instance (priority := 100) FloorRing.archimedean (α) [LinearOrderedField α] [F
 instance Units.instMulArchimedean (α) [OrderedCommMonoid α] [MulArchimedean α] :
     MulArchimedean αˣ :=
   ⟨fun x {_} h ↦ MulArchimedean.arch x.val h⟩
+
+instance WithBot.instArchimedean (α) [OrderedAddCommMonoid α] [Archimedean α] :
+    Archimedean (WithBot α) := by
+  constructor
+  intro x y hxy
+  induction y with
+  | bot => exact absurd hxy bot_le.not_lt
+  | coe y =>
+    induction x with
+    | bot => refine ⟨0, bot_le⟩
+    | coe x => simpa [← WithBot.coe_nsmul] using (Archimedean.arch x (by simpa using hxy))
+
+@[to_additive existing WithBot.instArchimedean]
+instance WithZero.instMulArchimedean (α) [OrderedCommMonoid α] [MulArchimedean α] :
+    MulArchimedean (WithZero α) := by
```

Comment:
Is it kosher to additivise `WithZero` to `WithBot`?

Reply:
I don't think so. Because with have `LinearOrderedCommGroupWithZero` and `LinearOrderedCommGroupWith**Top**`.

Reply:
Then can you drop the `to_additive`?